04_numeronym/willshen8/numeronyms.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	returnNumeronyms := []string{}
+	for _, val := range vals {
+		returnNumeronyms = append(returnNumeronyms, convertStringToNumeronym(val))
+	}
+	return returnNumeronyms
+}
+
+func convertStringToNumeronym(input string) string {
+	regEx := regexp.MustCompile(`[^a-zA-Z0-9]+`)
+	formattedInput := regEx.ReplaceAllString(input, "")
+
+	if len(formattedInput) <= 3 {
+		return formattedInput
+	}
+
+	firstChar := string(formattedInput[0])
+	lastChar := string(formattedInput[len(formattedInput)-1])
+	return firstChar + strconv.Itoa(len(formattedInput)-2) + lastChar
```

Comment:
Would suggest to use sprintf for this purpose as its easier.
`end:=len(input)
 return fmt.Sprintf("%c%d%c", input[0], end-2, input[end-1]))`
Something along these lines would remove the need for assigning multiple variables.

Reply:
Agree, this is neater approach.